core/src/main/java/cz/xtf/core/openshift/OpenShift.java

```diff
@@ -372,6 +444,38 @@ public boolean deletePods(String key, String value) {
 	public boolean deletePods(Map<String, String> labels) {
 		return pods().withLabels(labels).delete();
 	}
+	
+	/**
+	 * Retrieve pod containers
+	 * 
+	 * @param pod pod to retrieve in
+	 * @return List of containers of empty list if none ...
+	 */
+	public List<Container> getAllContainers(Pod pod) {
+		return Optional.ofNullable(pod.getSpec()).map(PodSpec::getContainers).orElse(new ArrayList<>());
+	}
+
+	/**
+	 * Retrieve any container from the given pod
+	 * 
+	 * @param pod Pod to retrieve from
+	 * @return One random container from the pod
+	 */
+	public Container getAnyContainer(Pod pod) {
+		List<Container> containers = getAllContainers(pod);
+		return containers.get(new Random().nextInt(containers.size()));
+	}
+	
+	public Container getContainerByName(Pod pod, String containerName) {
+		return getAllContainers(pod).stream()
+			.filter(c -> c.getName().equals(containerName))
+			.findFirst()
+			.orElseThrow(() -> new RuntimeException("Cannot find container with name " + containerName + " in pod " + pod.getMetadata().getName()));
+	}
+
+	private <R> Map<String, R> retrieveFromPodContainer(Pod pod, Function<Container, R> containerRetriever) {
```

Comment:
Please make a method name plural so it's consistent with return type (`retrieveFromPodContainer`). This way it seems it should be from one container.

Reply:
updated